apps/cli/src/commands/list.command.ts

```diff
@@ -127,20 +131,168 @@ export class ListTasksCommand extends Command {
 			await this.initializeCore(getProjectRoot(options.project));
 
 			// Get tasks from core
-			const result = await this.getTasks(options);
+			if (options.watch) {
+				await this.watchTasks(options);
+			} else {
+				const result = await this.getTasks(options);
 
-			// Store result for programmatic access
-			this.setLastResult(result);
+				// Store result for programmatic access
+				this.setLastResult(result);
 
-			// Display results
-			if (!options.silent) {
-				this.displayResults(result, options);
+				// Display results
+				if (!options.silent) {
+					this.displayResults(result, options);
+				}
 			}
 		} catch (error: any) {
 			displayError(error);
 		}
 	}
 
+	/**
+	 * Format a timestamp for display
+	 */
+	private formatSyncTime(date: Date): string {
+		return date.toLocaleTimeString('en-US', {
+			hour: '2-digit',
+			minute: '2-digit',
+			second: '2-digit',
+			hour12: true
+		});
+	}
+
+	/**
+	 * Display watch status footer
+	 */
+	private displayWatchFooter(
+		storageType: 'api' | 'file',
+		lastSync: Date
+	): void {
+		const syncTime = this.formatSyncTime(lastSync);
+		const source = storageType === 'api' ? 'Hamster Studio' : 'tasks.json';
+
+		console.log(chalk.dim(`\nWatching ${source} for changes...`));
+		console.log(chalk.gray(`Last synced: ${syncTime}`));
+	}
+
+	/**
+	 * Watch for changes and update list
+	 */
+	private async watchTasks(options: ListCommandOptions): Promise<void> {
+		// Initial render
+		let result = await this.getTasks(options);
+		let lastSync = new Date();
+
+		console.clear();
+		this.displayResults(result, options);
+
+		const storageType = result.storageType;
+
+		if (storageType === 'api') {
+			// API implementation using Supabase Realtime
+			const { AuthManager } = await import('@tm/core');
+			const authManager = AuthManager.getInstance();
+			const context = authManager.getContext();
+
+			if (!context?.briefId) {
+				console.warn(chalk.yellow('No brief context found for watching.'));
+				return;
+			}
+
+			const supabase = authManager.supabaseClient.getClient();
+
+			this.displayWatchFooter('api', lastSync);
+
+			const channel = supabase
```

Comment:
this can go in tm-core, doesn't make sense to have it here. under supabase client or a separate class